mesonbuild/compilers/compilers.py

```diff
@@ -851,7 +851,10 @@ def build_unix_rpath_args(self, build_dir, from_dir, rpath_paths, build_rpath, i
                 paths = padding
             else:
                 paths = paths + ':' + padding
-        args = ['-Wl,-rpath,' + paths]
+        args = []
+        if mesonlib.is_dragonflybsd():
+            args.append('-Wl,-z,origin')
```

Comment:
What does this do, exactly?

Reply:
it tells the linker that the elf needs to record $ORIGIN, without this ORIGIN is undefined and the rpath doesn't work at all

Reply:
Please add that as a comment :)